Improve get started docs and guide to working with notebooks

stichbury opened the pull request:
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.

## Description
Ready for review by @yetudada @datajoely @merelcht and anyone else interested!

Sorry, sorry, sorry, it looks to be a massive PR 🙀 

It's not as bad as it looks but I've renamed some pages and changes some sub-section names so I've had to modify a few files extra just to fix broken links.

I suggest you take a look mostly at the resulting output which I've put here:
https://stichbury.github.io/

> There are some conflicts now as a result of other PR submissions but I'll handle that in parallel with any review.

## Changelog 

* Revised the Introduction to make it short and sweet.
* Revised the Get Started section. Gone is "Hello Kedro". Gone are the installation pre-requisites (that's just part of the Install Kedro page now). Gone is the "Standalone use of the data catalog - woot woot" and GONE is the page on Kedro starters.
* I've reordered the create project material to put the project structure breakdown in the section that introduces key concepts and shorten the Iris tutorial to the bare minimum. I did add visualisation at this point though, to highlight Kedro Viz, as I felt it was coming far too late in the spaceflights tutorial and needed to be more prominent as a feature.
* Added a TL;DR page to Get Started which some people could probably just use as-is and ignore the rest of the section.
* Starters material has moved into a new section all about "Kedro project setup". Much of that section still needs review/revision but I have updated the Starters page so it reads more clearly.

I think that's about all, apart from some minor fixes (Github -> GitHub) and fixes to links for Slack.


## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes


<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2031"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

stichbury commented:
Ready for review by @yetudada @datajoely @merelcht and anyone else interested!

Sorry, sorry, sorry, it looks to be a massive PR 🙀 

It's not as bad as it looks but I've renamed some pages and changes some sub-section names so I've had to modify a few files extra just to fix broken links.

I suggest you take a look mostly at the resulting output which I've put here:
https://stichbury.github.io/

> There are some conflicts now as a result of other PR submissions but I'll handle that in parallel with any review.

## Changelog 

* Revised the Introduction to make it short and sweet.
* Revised the Get Started section. Gone is "Hello Kedro". Gone are the installation pre-requisites (that's just part of the Install Kedro page now). Gone is the "Standalone use of the data catalog - woot woot" and GONE is the page on Kedro starters.
* I've reordered the create project material to put the project structure breakdown in the section that introduces key concepts and shorten the Iris tutorial to the bare minimum.
* Added a TL;DR page to Get Started which some people could probably just use as-is and ignore the rest of the section.
* Starters material has moved into a new section all about "Kedro project setup". Much of that section still needs review/revision but I have updated the Starters page so it reads more clearly.

I think that's about all, apart from some minor fixes (Github -> GitHub) and fixes to links for Slack.

stichbury commented on docs/source/get_started/new_project.md:
````diff
@@ -36,21 +41,92 @@ The `output_dir` can be set to wherever you want to create the project. For exam
+| Report accuracy | Reports the accuracy of the predictions performed by the previous node.             | `report_accuracy`  |
+
+### Visualise the project
+
+This is a swift introduction to show how to visualise the project with Kedro Viz. See the [visualisation documentation](../visualisation/kedro-viz_visualisation) for more detail.
+
+In your terminal type the following:
 
 ```bash
-git init
+kedro viz
````
Ah, good point. I'll add Kedro-Viz to the `pandas-iris` starter as a dependency so they will have it.

datajoely commented on docs/source/get_started/kedro_concepts.md:
```diff
@@ -0,0 +1,74 @@
+
+## Node
+
+In Kedro, a node is a wrapper for a Python function that names the inputs and outputs of that function. Nodes are the building block of a pipeline, and the output of one node can be the input of another.
```
I pushed for this, I quite like it

stichbury replied:
"pure" == no side effects, right? So we mean stateless.

I think we should avoid too much complexity at this stage. I do however want to put a graphic on this page that combines the best of these two graphics (I'm still working on a prototype for Mackay)

<img width="756" alt="Screenshot 2022-11-18 at 10 55 49" src="https://user-images.githubusercontent.com/5180475/202760958-40d99ddf-8f69-4d51-88db-8ebf02c03b1d.png">
<img width="757" alt="Screenshot 2022-11-18 at 09 53 36" src="https://user-images.githubusercontent.com/5180475/202760967-0cfa22c2-de1c-4f49-bab1-18420e86693f.png">

noklam replied:
I like the pure Python function concepts but I also agree that introduce these term earlier could be a bit intimidating.

datajoely replied:
Yeah we could also just hyperlink to this one the word [pure](https://realpython.com/python-functional-programming/#:~:text=A%20pure%20function%20is%20a,to%20state%20or%20mutable%20data.)

stichbury replied:
I think I'll add a definition to the glossary and link to that rather than to the resource, which is great, but is itself a bit intimidating. I'll add the definition to Kedro terms and link to the resource from there, so it's a second order link for the keenest readers only :)

merelcht reviewed (approved):
Just some minor comments from me. This is a great improvement! The shorter pages make the docs much easier to follow 👍 ⭐

stichbury commented on docs/source/get_started/summary.md:
```diff
@@ -0,0 +1,33 @@
+    * or you can populate a new project with pre-built code, e.g. `kedro new --starter=pandas-iris` from a [range of starter projects](../kedro_project_setup/starters.md#list-of-official-starters)
+* Once you've created a project, you need to navigate to its project folder; you can then install its dependencies: `pip install -r src/requirements.txt`
+* To run the project: `kedro run`
+* To visualise the project: `kedro viz`
```
You're right, and @noklam raised this too. I am updating the `pandas-iris` starter to add kedro-viz to dependencies. It seems a bit odd to create a dependency to serve the docs, but I think it's an acceptable addition because many/most project users will want it. Make sense @merelcht ?

merelcht replied:
Yes it does, but if the user opts to create an empty project and not follow the starter instructions, then they still wouldn't have viz. I guess it's likely that most people would follow the starter instructions and not go into the empty project straight away, so it's very minor.

stichbury commented on docs/source/get_started/summary.md:
```diff
@@ -0,0 +1,33 @@
+
+
+## Kedro project creation
+* You can create a Kedro project:
```
I don't know tbh. I want to keep this page exceptionally simple, straightforward and clear. I think it complicates things. What I'll do is make a minor change to explain where the information came from (link each subhead) so they can go back to that page and find out more.

merelcht replied:
That sounds good!

jmholzer commented on docs/source/notebooks_and_ipython/kedro_and_notebooks.md:
````diff
@@ -0,0 +1,286 @@
+| `load_versions` | `Dict[str, str]` | A mapping of a dataset name to a specific dataset version (timestamp) for loading. Applies to versioned datasets only                                |
+| `pipeline_name` | `str`            | Name of the modular pipeline to run. Must be one of those returned by the `register_pipelines` function in `src/<package_name>/pipeline_registry.py` |
+
+## `%reload_kedro` line magic
+
+You can use `%reload_kedro` [line magic](https://ipython.readthedocs.io/en/stable/interactive/magics.html) within your Notebook to reload the Kedro variables (for example, if you need to update `catalog` following changes to your Data Catalog).
+
+You don't need to restart the kernel to reload the Kedro IPython extension and refresh the `catalog`, `context`, `pipelines` and `session` variables.
+
+`%reload_kedro` accepts optional keyword arguments `env` and `params`. For example, to use configuration environment `prod`:
+
+```ipython
+%reload_kedro --env=prod`
+```
+
+For more details, run `%reload_kedro?`.
````
This section is excellent! So well explained, excellent work 🚀

jmholzer reviewed (commented):
Amazing work! These docs are so well written, clear and easy to follow 🚀🚀.

I especially appreciate your work on notebooks.

I have left a few comments, none too major.

yetudada reviewed (commented):
I love these changes! I left minor comments.

stichbury commented on docs/source/get_started/kedro_concepts.md:
```diff
@@ -0,0 +1,66 @@
+It is time to introduce the most basic elements of Kedro. You can find further information about these and more advanced Kedro concepts in the [Kedro glossary](../resources/glossary.md).
+## Node
+
+In Kedro, a node is a wrapper for a [pure Python function](../resources/glossary.md#node) that names the inputs and outputs of that function. Nodes are the building block of a pipeline, and the output of one node can be the input of another.
```
I know! I didn't have this, and both Nok & Joel asked for it. I compromised with a link to the glossary that explains "pure" further, but I don't know that it helps simplify the experience. Let me see if I can find a way to satisfy both points of view 😆

stichbury commented:
All updated following reviews by @noklam @merelcht @yetudada and @jmholzer 

Thanks for your feedback. I've made loads of changes -- we're now at 97 commits for this PR 😆 

Please can I get a final approval to merge?

The latest and greatest is visible on https://stichbury.github.io/

noklam reviewed (approved):
Approved ⭐️ x100

jmholzer reviewed (approved):
Thanks so much for this! 🎉

stichbury commented:
* Revised the Introduction to make it short and sweet.
* Revised the Get Started section. Gone is "Hello Kedro". Gone are the installation pre-requisites (that's just part of the Install Kedro page now). Gone is the "Standalone use of the data catalog - woot woot" and GONE is the page on Kedro starters.
* Reordered the create project material to put the project structure breakdown in the section that introduces key concepts and shorten the Iris tutorial to the bare minimum. I did add visualisation at this point though, to highlight Kedro Viz, as I felt it was coming far too late in the spaceflights tutorial and needed to be more prominent as a feature.
* Added a TL;DR page to Get Started which some people could probably just use as-is and ignore the rest of the section.
* Starters material has moved into a new section all about "Kedro project setup". Much of that section still needs review/revision but I have updated the Starters page so it reads more clearly.
* Improved the Kedro-Viz page somewhat (still more to come for Plotly)
* Notebooks/IPython materials now merged and simplified
